src/Core/src/Handlers/Entry/EntryHandler.Android.cs

```diff
@@ -182,18 +183,38 @@ void OnFocusedChange(object? sender, FocusChangeEventArgs e)
 		void OnEditorAction(object? sender, EditorActionEventArgs e)
 		{
 			var returnType = VirtualView?.ReturnType;
+			bool handled = false;
 
-			if (returnType != null)
+			if (returnType is not null)
 			{
 				var currentInputImeFlag = returnType.Value.ToPlatform();
 
 				if (e.IsCompletedAction(currentInputImeFlag))
 				{
 					VirtualView?.Completed();
+
+					if (returnType == ReturnType.Done &&
+						e.ActionId == ImeAction.ImeNull &&
+						e.Event?.KeyCode == Keycode.Enter &&
+						e.Event?.Action == KeyEventActions.Up)
+					{
+						handled = true;
+					}
+				}
+				else if (e.ActionId == ImeAction.ImeNull
+					&& e.Event?.KeyCode == Keycode.Enter
+					&& e.Event?.Action == KeyEventActions.Down)
+				{
+					// If the user has indicated that they want this to be a "done" field
+					// then we will mark this as handled if we detect the enter key.
+					// The reason for doing this on Down is that this will now cause the "Up"
+					// to still be processed by this Editor action
+					// The `IsCompletedAction` above will process the `completed`
+					handled = true;
 				}
```

Comment:
A bit confused by here by the whole logic on handling on "Completed" and on "Up" vs. "Down" here - can you help me understand this a bit better?

Reply:
We want completed to fire when the user releases the enter key. 

By default if you don't mark this handled then Android will move the cursor to the next field as part of the down action but if the users has marked the return type as "Completed" then we want to keep the focus captured on the entry and then once they release the enter key we will process that as completed

Reply:
so I understand that as it pertains to Lines 192-203, but I'm a bit confused about the logic starting at the else statement on line 204.

if the action has not been completed, and enter is pressed down, then also handle it?